popgen/GvcfTajima.go

```diff
@@ -0,0 +1,140 @@
+package popgen
+
+import (
+	"github.com/vertgenlab/gonomics/vcf"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+	//DEBUG: "fmt"
+	"strings"
+	"math"
+)
+/*
+This file contains functions for calculating Tajima's D from a gVCF. More information on Tajima's D can be found at https://en.wikipedia.org/wiki/Tajima%27s_D
+*/
+
+//IndividualAllele is an internal struct for the Tajima's D calculation from gVCFs. It contains the "column" information of a gVCF block
+//In other words, an IndividualAllele represents the state of each segregating site for one allele in a gVCF block.
+type IndividualAllele struct {
+	sites [][]dna.Base
+} 
+
+//IndividualDist is a helper function for Tajima's D calculations that calculates the distance (the number of segregating sites that are different by state) between two IndividualAllele structs.
+func IndividualDist(a IndividualAllele, b IndividualAllele) int {
+	var answer int = 0
+	if len(a.sites) != len(b.sites) {
+		log.Fatalf("IndividualAllele sites lists must be the same length to calculate distance.")
+	}
+	for i := 0; i < len(a.sites); i++ {
+		if len(a.sites[i]) != len(b.sites[i]) {
```

Comment:
How do we feel about gaps?  Do we want a check where it fails if there are gaps?  Right now it counts them as a difference